### PR TITLE
Bug fix galore

### LINKED ISSUE #159: User gets prompted before closing tab on the response submitted page
Steps to reproduce:
1. Fill out a survey response
2. Submit the response
3. Wait for the response submitted page
4. Try to close the browser tab
5. See that you get prompted to close the tab

It should not prompt the user to close the tab after submitting successfully.

### PR DESCRIPTION
**Pull request by nick-dimeglio**
Essentially ran through a few known issues:

1) Weird behavior of small "X" buttons on the survey edit page is now fixed.
2) Event listener is properly removed on the survey respond page to prevent site from prompting user about not leaving after survey has been submitted.

Closes #159

### CODE REVIEW

File: packages/frontend/src/routes/survey/[slug]/respond/+page.svelte [hunk 1]
```diff
@@ -29,6 +29,12 @@
 				if (resp.value !== null) {
 					responderUuid = resp.value.responder_uuid;
 				}
+				//override the event listener then properly remove it below.
+				window.onbeforeunload = function () {
+					return;
+				}
```
Comment (dyc3) [inline packages/frontend/src/routes/survey/[slug]/respond/+page.svelte:35]: i would prefer it if there was some sort of "page nav" event we could listen for instead.

File: packages/frontend/src/routes/survey/[slug]/respond/+page.svelte [hunk 2]
```diff
@@ -29,6 +29,12 @@
...
+
+				window.removeEventListener('beforeunload', window.onbeforeunload);
```
Comment (dyc3) [inline packages/frontend/src/routes/survey/[slug]/respond/+page.svelte:37]: also, you do it twice?

File: packages/frontend/src/lib/ui/Button.svelte [hunk 3]
```diff
@@ -67,10 +67,12 @@
 		border: none;
 		transition: all $transition-duration ease-in-out;
 		margin: var(--margin, 0);
-
+		height: 100%;
+		
 		* {
 			transition: all $transition-duration ease-in-out;
 		}
+
```
Comment (dyc3) [inline packages/frontend/src/lib/ui/Button.svelte:75]: nit: extraneous newline

**Review by dyc3 — CHANGES_REQUESTED** [review]

File: packages/frontend/src/routes/survey/[slug]/respond/+page.svelte (hunk 1 above)
Comment (nick-dimeglio) [reply packages/frontend/src/routes/survey/[slug]/respond/+page.svelte:35]: If you have an event listener that you have in mind, I can change this, but as far as I can tell, onbeforeunload is the standard way to handle people leaving the page. A quick canvas online doesn't really show any other suitable event listener that will handle all of the possible page closures.

Here is a good link I found on it: https://stackoverflow.com/questions/15925251/trigger-an-event-when-user-navigates-away

**Comment by nick-dimeglio** [comment]
> 

So you'd like a test confirming that no dialog box appears once the survey is created?

**Comment by dyc3** [comment]
Yes

File: packages/frontend/src/routes/survey/[slug]/respond/+page.svelte (hunk 1 above)
Comment (nick-dimeglio) [reply packages/frontend/src/routes/survey/[slug]/respond/+page.svelte:35]: I looked into beforeNavigate and fiddled with it for an hour to try to get something up and working with it, but there seems to be huge issues with implementing it for some reason. I can try again during the lab today and explain what I mean better.

**Review by dyc3 — APPROVED** [review]

**Comment by dyc3** [comment]
Merging because the test failures are unrelated to the changes in this PR.

**Pull request merged**